spirv_glsl.cpp

```diff
@@ -4009,6 +4024,18 @@ string CompilerGLSL::constant_expression_vector(const SPIRConstant &c, uint32_t
 		}
 		break;
 
+	// Metal tessellation likes empty structs which are then constant expressions.
+	case SPIRType::Struct:
+		if (type.member_types.size() == 0)
```

Comment:
Struct constant expression are emitted by going through subconstants of SPIRConstant. Is this path ever hit?

Reply:
This condition is triggered in `shaders-msl/asm/tesc/hs-incorrect-base-type.asm.tesc` - one of the new test shaders.

Reply:
Alright I see now what's going on. The struct is empty, so subconstants is empty, so it falls down to the constant_expression_vector path. Please move this path to CompilerGLSL::constant_expression as an else if which would be clearer.

Reply:
Done in [spirv_glsl.cpp:3440](https://github.com/KhronosGroup/SPIRV-Cross/pull/1162/files#diff-09b7cdf207013ada0b4e38970bc27158R3440)